Osiris/Hacks/Misc.cpp

```diff
@@ -542,38 +540,45 @@ void Misc::HookRecorder(UserCmd* cmd) noexcept
 		}
 	}
 }
-
-void Misc::VisualizeRecorder() noexcept
+void Misc::visRecorder(ImDrawList* drawList) noexcept
 {
-	if (config->misc.visualizeRecorder)
-	{
-		/*This isnt working good, its rendering in not good position: https://streamable.com/imlwj8
-		if you have idea how to fix that, please let me know'*/
-
-		ImVec2 pos;
-		ImVec2 pos2;
+	if (!config->misc.visualizeRecorder)
+		return;
 
-		FrameContainer& recording = recorder.GetActiveRecording();
+	GameData::Lock lock;
+	const auto& localPlayerData = GameData::local();
 
-		for (int i = 1; i < recording.size(); i++)
+	if (!localPlayerData.exists || !localPlayerData.alive)
+		return;
+	
+	if (localPlayer->getAbsOrigin().distTo(startVec) < 100.f || recorder.IsRecordingActive())
+	{
+		if (!playback.IsPlaybackActive() && !recorder.IsRerecordingActive())
 		{
-			if (mWorldToScreen(movePositions[i], pos))
+			for (int i = 2; i < movePositions.size(); i++)
 			{
-
-				interfaces->surface->setDrawColor(255, 0, 0);
-				interfaces->surface->drawCircle(pos.x, pos.y, 1, 3);
+				if (ImVec2 spos; mWorldToScreen(startVec, spos))
+					drawList->AddCircle({ spos.x, spos.y }, 5, ImColor(0.f, 1.f, 0.f, 1.f), 0, 7.f);
+				if (ImVec2 cpos; mWorldToScreen(movePositions[i - 1], cpos))
+					if (ImVec2 cpos2; mWorldToScreen(movePositions[i], cpos2))
+						drawList->AddLine({ cpos.x, cpos.y }, { cpos2.x, cpos2.y }, ImColor(1.f, 0.f, 0.f, 1.f), 2.f);
 			}
 		}
-		if (mWorldToScreen(startVec, pos2))
+	}
+	if (playback.IsPlaybackActive())
+	{
+		const size_t current_playback_frame = playback.GetCurrentFrame();
+		for (int i = 2; i < movePositions.size(); i++)
```

Comment:
why do you do i = 2?, it should be 0

Reply:
because this use the i movePosition, and i-1 movePositions, and i didnt need the first movePositions, because this same with startVec

	if (ImVec2 cpos; worldToScreen(**movePositions[i - 1]**, cpos))
		if (ImVec2 cpos2; worldToScreen(**movePositions[i]**, cpos2))